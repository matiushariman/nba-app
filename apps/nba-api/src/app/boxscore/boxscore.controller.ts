import { Controller, Get, Param } from '@nestjs/common';

import { BoxscoreService } from './boxscore.service';

import type { GetBoxscoreByGameIdReq } from '@nba-app/types-nba';

@Controller('boxscore')
export class BoxscoreController {
  constructor(private readonly boxscoreService: BoxscoreService) {}

  @Get(':gameId')
  getBoxscoreByGameId(@Param() params: GetBoxscoreByGameIdReq) {
    return this.boxscoreService.getBoxscoreByGameId(params.gameId);
  }
}
