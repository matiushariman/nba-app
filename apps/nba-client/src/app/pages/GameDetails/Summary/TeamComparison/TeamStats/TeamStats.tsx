import TeamStatsChart from './TeamStatsChart';

import type { TeamStatsProps } from './TeamStats.types';

const TeamStats = ({ label, awayTeam, homeTeam }: TeamStatsProps) => (
  <div className="grid gap-4">
    <div>
      <p className="font-bold text-sm">{label}</p>
    </div>
    <div className="grid gap-3">
      <div>
        <TeamStatsChart
          teamTricode={awayTeam.teamTricode}
          percentage={awayTeam.percentage}
          secondaryText={awayTeam.secondaryText}
          primaryText={awayTeam.primaryText}
          color="away"
        />
      </div>
      <div>
        <TeamStatsChart
          teamTricode={homeTeam.teamTricode}
          percentage={homeTeam.percentage}
          secondaryText={homeTeam.secondaryText}
          primaryText={homeTeam.primaryText}
        />
      </div>
    </div>
  </div>
);

export default TeamStats;
