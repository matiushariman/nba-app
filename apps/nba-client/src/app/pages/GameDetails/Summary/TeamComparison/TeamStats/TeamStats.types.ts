import type { Team } from '@nba-app/types-nba';

type TeamStats = {
  readonly teamTricode: Team['teamTricode'];
  readonly percentage: number;
  readonly secondaryText?: string;
  readonly primaryText: string | number;
};

export interface TeamStatsProps {
  readonly label: string;
  readonly awayTeam: TeamStats;
  readonly homeTeam: TeamStats;
}
