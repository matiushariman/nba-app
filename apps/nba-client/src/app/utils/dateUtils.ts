import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
