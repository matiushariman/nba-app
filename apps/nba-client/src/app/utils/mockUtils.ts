export const shouldUseMock = process.env['NODE_ENV'] === 'development';
