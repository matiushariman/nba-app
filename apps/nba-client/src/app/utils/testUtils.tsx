import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
