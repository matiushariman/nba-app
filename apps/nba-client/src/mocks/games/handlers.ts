import { rest } from 'msw';
import { GetGamesRes, GetGameByIdRes } from '@nba-app/types-nba';

export const mockGames = {
  meta: {
    version: 1,
    request:
      'https://nba-prod-us-east-1-mediaops-stats.s3.amazonaws.com/NBA/liveData/scoreboard/todaysScoreboard_00.json',
    time: '2022-12-12 09:43:05.435',
    code: 200,
  },
  scoreboard: {
    gameDate: '2022-12-12',
    leagueId: '00',
    leagueName: 'National Basketball Association',
    games: [
      {
        gameId: '0022200405',
        gameCode: '20221212/OKCDAL',
        gameStatus: 2,
        gameStatusText: 'Half',
        period: 2,
        gameClock: 'PT00M00.00S',
        gameTimeUTC: '2022-12-13T01:30:00Z',
        gameEt: '2022-12-12T20:30:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612742,
          teamName: 'Mavericks',
          teamCity: 'Dallas',
          teamTricode: 'DAL',
          wins: 13,
          losses: 13,
          score: 66,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 5,
          periods: [
            { period: 1, periodType: 'REGULAR', score: 23 },
            { period: 2, periodType: 'REGULAR', score: 43 },
            { period: 3, periodType: 'REGULAR', score: 0 },
            { period: 4, periodType: 'REGULAR', score: 0 },
          ],
        },
        awayTeam: {
          teamId: 1610612760,
          teamName: 'Thunder',
          teamCity: 'Oklahoma City',
          teamTricode: 'OKC',
          wins: 11,
          losses: 15,
          score: 52,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 5,
          periods: [
            { period: 1, periodType: 'REGULAR', score: 28 },
            { period: 2, periodType: 'REGULAR', score: 24 },
            { period: 3, periodType: 'REGULAR', score: 0 },
            { period: 4, periodType: 'REGULAR', score: 0 },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1629029,
            name: 'Luka Doncic',
            jerseyNum: '77',
            position: 'GF',
            teamTricode: 'DAL',
            playerSlug: 'luka-doncic',
            points: 24,
            rebounds: 7,
            assists: 4,
          },
          awayLeaders: {
            personId: 1628983,
            name: 'Shai Gilgeous-Alexander',
            jerseyNum: '2',
            position: 'GF',
            teamTricode: 'OKC',
            playerSlug: 'shai-gilgeous-alexander',
            points: 15,
            rebounds: 1,
            assists: 2,
          },
        },
        pbOdds: { team: null, odds: 0, suspended: 0 },
      },
      {
        gameId: '0022200403',
        gameCode: '20221212/BKNWAS',
        gameStatus: 3,
        gameStatusText: 'Final',
        period: 4,
        gameClock: '',
        gameTimeUTC: '2022-12-13T00:00:00Z',
        gameEt: '2022-12-12T19:00:00Z',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612764,
          teamName: 'Wizards',
          teamCity: 'Washington',
          teamTricode: 'WAS',
          wins: 11,
          losses: 17,
          score: 100,
          seed: null,
          inBonus: null,
          timeoutsRemaining: 0,
          periods: [
            { period: 1, periodType: 'REGULAR', score: 25 },
            { period: 2, periodType: 'REGULAR', score: 30 },
            { period: 3, periodType: 'REGULAR', score: 28 },
            { period: 4, periodType: 'REGULAR', score: 17 },
          ],
        },
        awayTeam: {
          teamId: 1610612751,
          teamName: 'Nets',
          teamCity: 'Brooklyn',
          teamTricode: 'BKN',
          wins: 17,
          losses: 12,
          score: 112,
          seed: null,
          inBonus: null,
          timeoutsRemaining: 1,
          periods: [
            { period: 1, periodType: 'REGULAR', score: 29 },
            { period: 2, periodType: 'REGULAR', score: 33 },
            { period: 3, periodType: 'REGULAR', score: 32 },
            { period: 4, periodType: 'REGULAR', score: 18 },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 203115,
            name: 'Will Barton',
            jerseyNum: '5',
            position: 'G',
            teamTricode: 'WAS',
            playerSlug: null,
            points: 22,
            rebounds: 7,
            assists: 7,
          },
          awayLeaders: {
            personId: 201142,
            name: 'Kevin Durant',
            jerseyNum: '7',
            position: 'F',
            teamTricode: 'BKN',
            playerSlug: null,
            points: 30,
            rebounds: 9,
            assists: 6,
          },
        },
        pbOdds: { team: null, odds: 0, suspended: 0 },
      },
      {
        gameId: '0022200408',
        gameCode: '20221212/BOSLAC',
        gameStatus: 1,
        gameStatusText: '10:30 pm ET',
        period: 0,
        gameClock: '',
        gameTimeUTC: '2022-12-13T03:30:00Z',
        gameEt: '2022-12-12T22:30:00Z',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612746,
          teamName: 'Clippers',
          teamCity: 'LA',
          teamTricode: 'LAC',
          wins: 15,
          losses: 13,
          score: 0,
          seed: null,
          inBonus: null,
          timeoutsRemaining: 0,
          periods: [
            { period: 1, periodType: 'REGULAR', score: 0 },
            { period: 2, periodType: 'REGULAR', score: 0 },
            { period: 3, periodType: 'REGULAR', score: 0 },
            { period: 4, periodType: 'REGULAR', score: 0 },
          ],
        },
        awayTeam: {
          teamId: 1610612738,
          teamName: 'Celtics',
          teamCity: 'Boston',
          teamTricode: 'BOS',
          wins: 21,
          losses: 6,
          score: 0,
          seed: null,
          inBonus: null,
          timeoutsRemaining: 0,
          periods: [
            { period: 1, periodType: 'REGULAR', score: 0 },
            { period: 2, periodType: 'REGULAR', score: 0 },
            { period: 3, periodType: 'REGULAR', score: 0 },
            { period: 4, periodType: 'REGULAR', score: 0 },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 0,
            name: '',
            jerseyNum: '',
            position: '',
            teamTricode: 'LAC',
            playerSlug: null,
            points: 0,
            rebounds: 0,
            assists: 0,
          },
          awayLeaders: {
            personId: 0,
            name: '',
            jerseyNum: '',
            position: '',
            teamTricode: 'BOS',
            playerSlug: null,
            points: 0,
            rebounds: 0,
            assists: 0,
          },
        },
        pbOdds: { team: null, odds: 0, suspended: 0 },
      },
    ],
  },
};

export const gamesHandlers = [
  rest.get<GetGameByIdRes>('/api/games/:gameId', (req, res, ctx) => {
    const { gameId } = req.params;
    const game = mockGames.scoreboard.games.find(
      (findGame) => findGame.gameId === gameId
    );
    return res(ctx.json(game));
  }),
  rest.get<GetGamesRes>('/api/games', (req, res, ctx) => {
    return res(ctx.json(mockGames));
  }),
];
