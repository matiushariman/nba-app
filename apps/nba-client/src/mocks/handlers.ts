import { boxscoreHandlers } from './boxscore/handlers';
import { gamesHandlers } from './games/handlers';

export const handlers = [...boxscoreHandlers, ...gamesHandlers];
