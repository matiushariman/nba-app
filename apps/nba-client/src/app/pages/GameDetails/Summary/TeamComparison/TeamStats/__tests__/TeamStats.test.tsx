import TeamStats from '..';

import { render } from '../../../../../../utils/testUtils';
import { mockBoxcore } from '../../../../../../../mocks/boxscore/handlers';

const { awayTeam, homeTeam } = mockBoxcore[0].game;

describe('pages/GameDetails/Summary/TeamComparison/TeamStats', () => {
  it('should render TeamStatsChart that matches snapshot', () => {
    const { baseElement } = render(
      <TeamStats
        label="Field Goals"
        awayTeam={{
          teamTricode: awayTeam.teamTricode,
          primaryText: '100/100',
          percentage: 100,
        }}
        homeTeam={{
          teamTricode: homeTeam.teamTricode,
          primaryText: '100/100',
          percentage: 100,
        }}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
