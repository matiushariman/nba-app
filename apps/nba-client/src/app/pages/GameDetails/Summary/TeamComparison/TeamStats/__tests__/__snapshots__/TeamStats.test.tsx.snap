// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/TeamComparison/TeamStats should render TeamStatsChart that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="grid gap-4"
    >
      <div>
        <p
          class="font-bold text-sm"
        >
          Field Goals
        </p>
      </div>
      <div
        class="grid gap-3"
      >
        <div>
          <div>
            <div
              class="flex items-center gap-2"
            >
              <div
                class="flex-grow"
              >
                <p
                  class="text-sm"
                >
                  OKC
                </p>
              </div>
              <div>
                <p
                  class="font-bold text-lg"
                >
                  100/100
                </p>
              </div>
            </div>
            <div
              class="relative h-4"
            >
              <div
                class="absolute h-4 bg-blue-800 z-10"
                style="width: 100%;"
              />
              <div
                class="absolute h-4 bg-gray-100 w-full"
              />
            </div>
          </div>
        </div>
        <div>
          <div>
            <div
              class="flex items-center gap-2"
            >
              <div
                class="flex-grow"
              >
                <p
                  class="text-sm"
                >
                  DAL
                </p>
              </div>
              <div>
                <p
                  class="font-bold text-lg"
                >
                  100/100
                </p>
              </div>
            </div>
            <div
              class="relative h-4"
            >
              <div
                class="absolute h-4 bg-red-800 z-10"
                style="width: 100%;"
              />
              <div
                class="absolute h-4 bg-gray-100 w-full"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
